common/src/main/kotlin/io/homeassistant/companion/android/di/NetworkModule.kt

```diff
@@ -0,0 +1,48 @@
+package io.homeassistant.companion.android.di
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.wifi.WifiManager
+import androidx.core.content.getSystemService
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import io.homeassistant.companion.android.common.data.network.NetworkHelper
+import io.homeassistant.companion.android.common.data.network.NetworkHelperImpl
+import io.homeassistant.companion.android.common.data.network.NetworkStatusMonitor
+import io.homeassistant.companion.android.common.data.network.NetworkStatusMonitorImpl
+import io.homeassistant.companion.android.common.data.network.WifiHelper
+import io.homeassistant.companion.android.common.data.network.WifiHelperImpl
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal abstract class NetworkModule {
+
+    companion object {
+        @Provides
+        @Singleton
+        fun provideConnectivityManager(@ApplicationContext appContext: Context) =
+            appContext.getSystemService<ConnectivityManager>()!!
```

Comment:
Replace with checkNotNull and explicit error message even if it should never happen.